components/event-sources/reconciler/object/peerauthentication.go

```diff
@@ -0,0 +1,52 @@
+package object
+
+import (
+	securityv1beta1apis "istio.io/api/security/v1beta1"
+	istiov1beta1apis "istio.io/api/type/v1beta1"
+	securityv1beta1 "istio.io/client-go/pkg/apis/security/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func NewPeerAuthentication(ns, name string, opts ...ObjectOption) *securityv1beta1.PeerAuthentication {
+	s := &securityv1beta1.PeerAuthentication{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: ns,
+			Name:      name,
+		},
+	}
+	for _, opt := range opts {
+		opt(s)
+	}
+	return s
+}
+
+// WithPermissiveModeDeprecated sets the mTLS mode of the PeerAuthentication to Permissive for the given port
```

Comment:
To mark this function as deprecated please check this guide: https://github.com/golang/go/wiki/Deprecated.

Reply:
Solved